### PR TITLE
CI pre-commit autoupdate

### PR DESCRIPTION
**Pull request by MarcoGorelli**
the automated one didn't run, maybe because the diff is too large?

this touches a lot of files, but it's mostly just `black` removing newlines

### CODE REVIEW

**Comment by topper-123** [comment]
> but it's mostly just `black` removing newlines

Is there anything else? I cab only see reoval of newlines, but there are a lot, so may I'm missing something?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -125,7 +125,7 @@ repos:
         language: python
         require_serial: true
         types_or: [python, pyi]
-        additional_dependencies: [black==22.10.0]
```
Comment (rhshadrach) [inline .pre-commit-config.yaml:128]: Update environment.yml too?

**Review by rhshadrach — CHANGES_REQUESTED** [review]

**Comment by MarcoGorelli** [comment]
> Could also upgrade autotyping to 23.2.0 (pyright as well but that will probably require some code changes).

I tried this but got
```
IndexError: string index out of range

Failed to codemod /home/marcogorelli/pandas-dev/pandas/core/arrays/categorical.py
```
I'll see if this needs reporting to them

> Is there anything else? I cab only see removal of newlines, but there are a lot, so may I'm missing something?

I don't think so, but perhaps let's keep that to a separate PR

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by MarcoGorelli** [comment]
thanks for reviewing!

**Pull request merged**